test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -420,35 +420,70 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 	})
 
-	// TODO: we have a dependency between tests(that in general bad practice, but saves us some tests run time),
-	// once we will want to run tests in the random order or without failFast we will need to refactor tests
 	Context("Updating of nodeSelector parameter and node labels", func() {
 		var mcp *machineconfigv1.MachineConfigPool
 		var newCnfNode *corev1.Node
 
 		newRole := "worker-test"
 		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+		var labelsDeletion = false
 		newNodeSelector := map[string]string{newLabel: ""}
 
-		testutils.BeforeAll(func() {
+		//fetch the latest profile
+		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		oldMcpSelector := make(map[string]string)
+
+		//fetch existing MCP Selector if exists in profile
+		if profile.Spec.MachineConfigPoolSelector != nil {
+			oldMcpSelector = profile.Spec.DeepCopy().MachineConfigPoolSelector
+		}
+		var removeLabels = func(nodeSelector map[string]string, targetNode *corev1.Node) error {
+			for l := range nodeSelector {
+				delete(targetNode.Labels, l)
+			}
+			label, err := json.Marshal(targetNode.Labels)
+			if err != nil {
+				return err
```

Comment:
mixing `return err` and `Expect*` is a smell. Let's pick either and be consistent within the single function

Reply:
additionally, this is fragile. A bit better:
1. clone the labels of the `targetNode`
2. remove the labels as needed
3. marshal the new labels
4. if and only if we succeeded this far, update the `targetNode` object (or possibly a `DeepCopy()` of it) and do the Patch operation

Reply:
@fromanirh I have updated the as per above requirements.